lib/regexp_parser/token.rb

```diff
@@ -12,6 +12,12 @@ class Regexp
   ].freeze
 
   Token = Struct.new(*TOKEN_KEYS) do
+    def initialize(*)
+      super
+
+      @previous, @next = nil
```

Comment:
Would it be more clear to do `@previous = @next = nil`? I realize it's the same end result, but I feel like `@next` is only really kind of being initialized as a side effect of it appearing in the statement, rather than explicitly due to an assignment.